### ISSUE TITLE
Regression in 9.4.20 (from 9.1.19) when creating ServletContextHandler with SECURITY flag and then replacing the SecurityHander

### ISSUE THREAD

**Issue by tbroyer** [eclipse/jetty.project#4009]
In our app, using an embedded server, this used to work with Jetty 9.4.19:

```java
    final ServletContextHandler context = new ServletContextHandler(SESSIONS | SECURITY | GZIP);
    // …
    context.setSecurityHandler(new SimpleSecurityHandler(authenticator));
    // …
    ServletHolder resources =
        new ServletHolder(
            "resources",
            new DefaultServlet(
                new ResourceService() {
                  @Override
                  @SuppressWarnings("squid:S00112")
                  protected void notFound(HttpServletRequest request, HttpServletResponse response)
                      throws IOException {
                    // For use with history.pushState: return the index for everything unknown.
                    // XXX: filter only extension-less (and/or with .html extension) paths?
                    try {
                      response.setHeader(HttpHeaders.CONTENT_LOCATION, "/");
                      request.getRequestDispatcher("/").forward(request, response);
                    } catch (ServletException e) {
                      throw new RuntimeException(e);
                    }
                  }
                }));
    resources.setInitParameter("dirAllowed", "false");
    context.addServlet(resources, "/*");
```

but broke with 9.4.20 in that our `SimpleSecurityHandler` wasn't called for `/index.html` on the forwarded dispatch, when requesting a `/foo/bar/whatever` URL.

I traced this to this line: https://github.com/eclipse/jetty.project/blame/jetty-9.4.20.v20190813/jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java#L612

This should probably use `_securityHandler` rather than `_sessionHandler` here.

Fortunately, in our case, the workaround is easy: remove the `SECURITY` flag; which is probably a good idea anyway.

**Comment by joakime** [comment]
@janbartel can you take a look at this?

**Comment by janbartel** [comment]
Yup, looks like a bug alright. See PR #4012 .

**Comment by janbartel** [comment]
Fixed for 9.4.21 and beyond.
